### ISSUE TITLE
Search results page

### ISSUE THREAD

**Issue by timpaul** [alphagov/govuk-design-system-backlog#189]
# What
How to display a list of search results

**Comment by devansXD** [comment]
It feels like this is potentially quite a large area and covers many different types of results, namely:

- Information type and breakdown
- Raw data presentation
 - Tabular or list based structure
- Filter behaviour and positioning
- Sort behaviour and positioning
- Action a user can take eg: pick a thing/multiple things and doing something else

**Comment by titlescreen** [comment]
I agree with @devansXD, the type of search results may well depend on 

-  the type of search carried out
-  the number of expected results
-  information needed with those results to make a decision 
-  are they using the basic or progressively enhanced version

As such, I can't comment on all the search result pages, and perhaps we could consider breaking this down into smaller more achievable items in the backlog then standardise them later?

# Progressively enhanced version

In the two services, we made on the DfE claim project we were working primarily on a school search. The progressively enhanced version was a more simple decision as we used accessible auto-complete (https://github.com/alphagov/accessible-autocomplete) with a couple of tweaks to increase the contrast on the search options and pull in some additional information about the schools. 

One of the user research participants (T15) had been using JAWS since 2002 and at the time of testing, they were on a Windows XP Machine. Generally speaking accessible autocomplete worked well for them.

> Once they had started typing it was easy for them to select options from the dropdown

> Possible issue, when changing the answer on this question as it remembered the response and they didn't know that the input was pre-filled



![image](https://user-images.githubusercontent.com/13239597/74223009-aafd4680-4cad-11ea-8bb5-996cf364bfcb.png)

# Non-javascript version

For the non-javascript search, we opted to return the search results as a list of radio buttons. This tested well for all users it just involved a bit of scrolling however, that seemed like preferable to a more dense list that was potentially harder to read.

![image](https://user-images.githubusercontent.com/13239597/74223416-8e154300-4cae-11ea-9ac6-95fa88b3ca20.png)

# Support issues related to our school search

Users got in touch with support when they:

- were taken to an ineligible page as their school was not included in the pilot
- they could not find the school because it had changed status (e.g. become an academy) and/or slightly changed its name

There was also an issue with apostrophes within the school name e.g. St Mary's or St Marys. Whilst it was possible to do something on the search results page to include both schools with and without apostrophes a more simple solution was to allow them to search via postcode.

**Comment by laura-biscuits** [comment]
Is there a separate backlog item (that I can't find) for when a search has no results to return and telling a user what to try instead?

**Comment by domoscargin** [comment]
Hi @laura-biscuits, thanks to @adyhoran1 there is now: https://github.com/alphagov/govuk-design-system-backlog/issues/264

**Comment by CharlotteDowns** [comment]
We've just [added some guidance on how to 'share findings about your users' with us](https://design-system.service.gov.uk/community/share-research-findings/) 📝. This will help us learn more about how your users understand search results within your service.
